Added Corrections and Notices component (#3524)
ONSdigital/design-system: src/components/details-panel/_macro-options.md

| Name         | Type                 | Required | Description                                             |
| ------------ | -------------------- | -------- | ------------------------------------------------------- |
| title        | string               | true     | Title for the details panel                             |

> | DetailsItems | array`<DetailsItem>` | true     | Settings for the array of [Details Item](#DetailsItem)  |

Reviewer: ```suggestion
| DetailsItems | array`<DetailsItem>` | true     | Settings for the array of [Details Items](#DetailsItem)  |
```

> | DetailsItems | array`<DetailsItem>` | true     | Settings for the array of [Details Items](#DetailsItem)  |

Reviewer: ```suggestion
| DetailsItems | array`<DetailsItem>` | true     | Settings for the array of [Details Items](#DetailsItem) |
```

| Name   | Type           | Required | Description                                             |
| ------ | -------------- | -------- | ------------------------------------------------------- |

> | id     | string         | true     | HTML `id` attribute for the details panel               |

Reviewer: Is ID required?

> | title  | string         | true     | Title of the component                                  |

Reviewer: I think "Title for the details panel" is a bit more descriptive

| groups | array`<group>` | true     | Settings for the array of [group items](#group)         |
| open   | boolean        | false    | Forces the details panel to be open when the page loads |

> ## group

Reviewer: ```suggestion
## Group
```

> ## Group

Reviewer: We no longer have multiple groups so I think we can rename this `detailsItems` or something

> | Name        | Type           | Required                                | Description                                  |
> | ----------- | -------------- | --------------------------------------- | -------------------------------------------- |
> | text        | string         | true                                    | Name of the item                             |
> | date        | `Object<Date>` | true                                    | An object for [Date](#date) set for the item |
> | description | string         | true                                    | Description for the item                     |
> | url         | string         | false(unless text is set to Correction) | Link to view superseded version              |

Reviewer: ```suggestion
| Name        | Type           | Required                                | Description                                  |
| ----------- | -------------- | --------------------------------------- | -------------------------------------------- |
| text        | string         | true                                     | Name of the item                             |
| date        | `Object<Date>` | true                                     | An object for [Date](#date) set for the item |
| description | string         | true                                     | Description for the item                     |
| url         | string         | false (unless text is set to Correction) | Link to view superseded version              |
```

> | url         | string         | false (unless text is set to Correction) | Link to view superseded version              |

Reviewer: Formatting needs fixing

> #### Date

Reviewer: ```suggestion
## Date
```

| Name   | Type           | Required | Description                                             |
| ------ | -------------- | -------- | ------------------------------------------------------- |
| id     | string         | true     | HTML `id` attribute for the details panel               |
| groups | array`<group>` | true     | Settings for the array of [group items](#group)         |
| open   | boolean        | false    | Forces the details panel to be open when the page loads |

## group

| Name       | Type               | Required | Description                                                                                                                                                                                                                                        |
| ---------- | ------------------ | -------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | title      | string             | true     | Heading for the groupItem and also title of the the details panel when there is one item in groupItems. <br> Example: "Correction" (for one item),"Corrections"(for multiple items) "Corrections and Notices" (multiple items and multiple groups) |

Reviewer: I think we discussed having a title and a plural title that we would switch between based on the number of items

| groupItems | `array<groupItem>` | true     | Settings for the array of [group items](#groupItem) within the group                                                                                                                                                                               |

## groupItem

| Name | Type           | Required                                       | Description                                             |
| ---- | -------------- | ---------------------------------------------- | ------------------------------------------------------- |
| date | `Object<Date>` | true                                           | An object for [Date](#date) set for the correction item |
| text | string         | true                                           | Description of the correction item                      |

> | url  | string         | false(unless title is set to Correction) | Link to view superseded version                         |

Reviewer: This isnt fully formatted. Are your commit hooks set up?

Author: I ran yarn install again. I think they are setup now.   
![Screenshot 2025-04-08 at 15 45 17](https://github.com/user-attachments/assets/3e05f144-8058-4d00-9c15-e92116382301)

> | url  | string         | falue(unless singleTitle is set to Correction) | Link to view superseded version                         |

Reviewer 2: ```suggestion
| url  | string         | false(unless title is set to Correction) | Link to view superseded version                         |
```
